SonarQube.TeamBuild.PreProcessor/AnalysisConfigGenerator.cs

```diff
@@ -18,11 +18,16 @@ public static class AnalysisConfigGenerator
         /// Combines the various configuration options into the AnalysisConfig file
         /// used by the build and post-processor. Saves the file and returns the config instance.
         /// </summary>
-        /// <returns></returns>
+        /// <param name="args">Processed command line arguments supplied the user</param>
+        /// <param name="buildSettings">Build environment settings</param>
+        /// <param name="serverProperties">Analysis properties downloaded from the SonarQube server</param>
+        /// <param name="analyzerSettings">Specifies the Roslyn analyzers to use</param>
+        /// <param name="programmaticProperties">Any additional programmatically set analysis properties. Any user-specified values will take priority</param>
```

Comment:
I had hoped that I'd just be able to have the BuildWrapperInstaller directly update the AnalysisConfig before it is saved. However, it turned out to be a bit more complicated since all of the logic to created the config is centralised in this generator class.
The least disruptive and cleanest change I could make at the moment was to add a general-purpose "programmatic settings" parameter that is taken into account when generating the config file. The config generator doesn't know or care where these values come from, it just needs to know that the have lower precedence than settings provided on the command line.